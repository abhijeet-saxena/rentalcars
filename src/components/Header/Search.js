import React, { useRef, useEffect, useState } from 'react';
import { Flex } from '../../common/components';
import { colors } from '../../common/helper';
import { debounce, times } from '../../common/helper';
import { SuggestionCard } from '../../common/components';
import {
    SearchContainer,
    FilterContainer,
    DateTime,
    InputContainer,
    Autocomplete,
    Suggestions,
    Time,
} from './styles';

const Search = () => {
    const [suggestions, setSuggestions] = useState([]);
    const [loading, setLoading] = useState(false);
    const search = useRef(null);

    useEffect(() => {
        document.addEventListener('click', (e) => {
            setSuggestions([]);
        });

        return document.removeEventListener('click', (e) => {
            setSuggestions([]);
        });
    }, []);

    const autoComplete = ({ target: { value } }) => {
        const searchText = value.trim();
        if (searchText === '' || searchText.length < 2) setSuggestions([]);
        else {
            setLoading(true);
            fetch(
                `https://www.rentalcars.com/FTSAutocomplete.do?solrIndex=fts_en&solrRows=6&solrTerm=${searchText}`,
            )
                .then((res) => res.json())
                .then((data) => setSuggestions([...data.results.docs]))
                .catch((err) => console.log(err))
                .finally(() => setLoading(false));
        }
    };

    const setValue = (val) => {
        search.current.value = val;
        setSuggestions([]);
    };

    return (
        <SearchContainer data-testid="search">
            <h2>Let's find your ideal car</h2>
            <FilterContainer data-testid="filters">
                <Flex alignItems="center" width="auto">
                    <input type="checkbox" id="diff-location"></input>
                    <label htmlFor="diff-location">
                        Drop car off at different location
                    </label>
                </Flex>
                <Flex alignItems="center" width="auto">
                    <input
                        type="checkbox"
                        defaultChecked={true}
                        id="driver-age"
                    ></input>
                    <label htmlFor="driver-age">Driver aged 30 â€“ 65 ?</label>
                </Flex>
            </FilterContainer>
            <InputContainer data-testid="inputs">
                <Autocomplete>
                    <input
                        type="text"
                        placeholder="Pick-Up Location"
                        onChange={debounce(autoComplete)}
                        ref={search}
                        aria-label="Enter Location"
                    ></input>
                    {suggestions.length > 0 && (
                        <Suggestions
                            id="suggestions"
                            role="alert"
                            aria-live="assertive"
                            aria-label={`Search results for ${
                                search.current.value
                            }. ${suggestions.reduce(
                                (acc, item) => (acc += `${item.name}, `),
                                '',
                            )}`}
                        >
                            {suggestions.map((item) => (
                                <SuggestionCard
                                    details={item}
                                    key={`${item.name}+${item.index}`}
                                    setValue={setValue}
                                ></SuggestionCard>
                            ))}
                        </Suggestions>
                    )}
                    {loading && (
                        <svg
                            fill={colors.textBlue}
                            viewBox="0 0 24 24"
                            height="20px"
                            xmlns="http://www.w3.org/2000/svg"
                        >
                            <path d="M13 2.05v3.03c3.39.49 6 3.39 6 6.92 0 .9-.18 1.75-.48 2.54l2.6 1.53c.56-1.24.88-2.62.88-4.07 0-5.18-3.95-9.45-9-9.95zM12 19c-3.87 0-7-3.13-7-7 0-3.53 2.61-6.43 6-6.92V2.05c-5.06.5-9 4.76-9 9.95 0 5.52 4.47 10 9.99 10 3.31 0 6.24-1.61 8.06-4.09l-2.6-1.53A6.95 6.95 0 0 1 12 19z" />
                        </svg>
                    )}
                </Autocomplete>
                <Flex gap="0.75rem">
                    <DateTime>
                        <input
                            type="date"
                            id="pickup-date"
                            name="pickup-date"
                            defaultValue="2021-05-28"
                            aria-label="Enter Pickup date"
                        />
                        <Time>
                            <select
                                name="pickup-time"
                                defaultValue="10:30"
                                aria-label="Select Pickup Time"
                            >
                                {times.map((item) => (
                                    <option value={item} key={item}>
                                        {item}
                                    </option>
                                ))}
                            </select>
                        </Time>
                    </DateTime>
                    <DateTime>
                        <input
                            type="date"
                            id="drop-date"
                            name="drop-date"
                            defaultValue="2021-06-01"
                            aria-label="Enter Drop date"
                        />
                        <Time>
                            <select
                                name="drop-time"
                                defaultValue="15:30"
                                aria-label="Select Drop Time"
                            >
                                {times.map((item) => (
                                    <option value={item} key={item}>
                                        {item}
                                    </option>
                                ))}
                            </select>
                        </Time>
                    </DateTime>
                </Flex>
                <button type="button">Search</button>
            </InputContainer>
        </SearchContainer>
    );
};

export default Search;
