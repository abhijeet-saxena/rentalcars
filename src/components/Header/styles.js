import { Flex } from '../../common/components';
import { colors, rotate } from '../../common/helper';
import styled from 'styled-components';

export const HeaderContainer = styled.header`
    width: 100%;
    max-width: 1142px;
    padding: 1rem 1rem;
    color: ${colors.textLight};
    > h1 {
        margin: 2rem 0;
        text-align: center;
    }
`;

export const NavContainer = styled.nav`
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;
    > .menu {
        cursor: pointer;
        > img {
            width: 1.5rem;
            height: 1.5rem;
            border-radius: 50%;
        }
        > button {
            padding: 0px 16px;
            color: ${colors.primaryBlue};
            background: ${colors.textLight};
        }
    }
    .menu {
        display: none;
    }
    .hamburger {
        cursor: pointer;
        background: none;
        padding: 0;
    }

    @media (min-width: 768px) {
        .hamburger {
            display: none;
        }
        .menu {
            display: flex;
        }
    }
`;

export const SearchContainer = styled(Flex)`
    background: ${colors.secondaryYellow};
    flex-direction: column;
    padding: 1rem;
    border-radius: 8px;
    > h3 {
        color: black;
    }
    & button {
        padding: 1rem 3rem;
        color: ${colors.textLight};
        background: ${colors.secondaryGreen};
    }
`;

export const FilterContainer = styled(Flex)`
    gap: 1rem;
    flex-direction: column;
    margin: 1rem 0;
    & input {
        width: 20px;
        height: 20px;
        margin-right: 0.5rem;
    }
    & label {
        font-size: 14px;
    }

    @media (min-width: 768px) {
        gap: 2rem;
        flex-direction: row;
    }
`;

export const InputContainer = styled(Flex)`
    flex-direction: column;
    gap: 0.75rem;

    @media (min-width: 768px) {
        flex-direction: row;
    }
`;

export const Autocomplete = styled(Flex)`
    position: relative;
    & input {
        padding-left: 2rem;
        min-width: 100px;
    }
    &:after {
        content: url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="1em"><path d="m20.75 3.709-7.377 17.79-1.638-8.187a.75.75 0 0 0-.588-.588L2.96 11.086l17.79-7.377zm-.574-1.386L2.385 9.7a1.5 1.5 0 0 0 .28 2.856l8.188 1.638-.588-.588 1.638 8.188a1.5 1.5 0 0 0 2.856.28l7.377-17.79a1.5 1.5 0 0 0-1.96-1.96z"/></svg>');
        position: absolute;
        left: 0.5rem;
        top: 1rem;
    }
    > svg {
        position: absolute;
        right: 0.5rem;
        top: 0.75rem;
        animation: ${rotate} 1.5s linear infinite;
    }
`;

export const Suggestions = styled(Flex)`
    position: absolute;
    z-index: 1;
    top: 100%;
    flex-direction: column;
    background: ${colors.textLight};
    border: solid 1px ${colors.textDark};
    border-radius: 4px;
    cursor: pointer;
    > div {
        height: 54px;
        padding: 0.5rem;
        border-bottom: solid 1px ${colors.border};
        &:hover {
            background: ${colors.primaryBlueLight};
        }
    }
`;

export const DateTime = styled(Flex)`
    flex-direction: column;
    gap: 0.75rem;
    > input {
        width: auto;
        padding: 0.5rem;
    }
    & select {
        width: 100%;
    }
    overflow: hidden;

    @media (min-width: 768px) {
        flex-direction: row;
        gap: 0;
        border-radius: 4px;
        & input,
        select {
            border-radius: 0px;
            width: auto;
        }
        & select {
            border-left: solid 1px ${colors.border};
        }
    }
`;

export const Time = styled.div`
    position: relative;
    & select {
        padding-left: 1.75rem;
    }
    &:after {
        content: url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="1em"><path d="M21.75 12c0 5.385-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12 6.615 2.25 12 2.25s9.75 4.365 9.75 9.75zm1.5 0C23.25 5.787 18.213.75 12 .75S.75 5.787.75 12 5.787 23.25 12 23.25 23.25 18.213 23.25 12zm-10.5 0V8.25a.75.75 0 0 0-1.5 0V12a.75.75 0 0 0 1.5 0zm-1.28.53 4.687 4.688a.75.75 0 0 0 1.06-1.06L12.53 11.47a.75.75 0 0 0-1.06 1.06z"/></svg>');
        position: absolute;
        left: 0.5rem;
        top: 0.9rem;
    }
`;
