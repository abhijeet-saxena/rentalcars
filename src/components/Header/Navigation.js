import React from 'react';
import { NavContainer } from './styles';
import { Flex, Box, colors } from '../../common/components';
import logo from '../../assets/images/rc-logo.svg';
import gb from '../../assets/images/gb.png';

const Navigation = () => {
    return (
        <NavContainer>
            <a href="/">
                <img src={logo} alt="Rentalcars.com Logo"></img>
            </a>
            <Flex
                width="auto"
                gap="1.5rem"
                className="menu"
                alignItems="center"
            >
                <span style={{ color: colors.textLight }}>INR</span>
                <img src={gb} alt="gb"></img>
                <button>Manage Booking</button>
            </Flex>

            <Box className="hamburger">
                <svg viewBox="0 0 24 24" width="20">
                    <path
                        stroke="currentColor"
                        fill="currentColor"
                        d="M2.25 18.753h19.5a.75.75 0 0 0 0-1.5H2.25a.75.75 0 0 0 0 1.5zm0-6h19.5a.75.75 0 0 0 0-1.5H2.25a.75.75 0 0 0 0 1.5zm0-6h19.5a.75.75 0 0 0 0-1.5H2.25a.75.75 0 0 0 0 1.5z"
                    />
                </svg>
            </Box>
        </NavContainer>
    );
};

export default Navigation;
