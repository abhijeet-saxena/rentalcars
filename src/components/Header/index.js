import React from 'react';
import { Flex } from '../../common/components';
import { colors } from '../../common/helper';
import { HeaderContainer } from './styles';
import Navigation from './navigation';
import Search from './search';

const Header = () => {
    return (
        <Flex background={colors.primaryBlue} justifyContent="center">
            <HeaderContainer data-testid="heading">
                <Navigation></Navigation>
                <h1>Car Hire â€“ Search, Compare &amp; Save</h1>
                <Search></Search>
            </HeaderContainer>
        </Flex>
    );
};

export default Header;
