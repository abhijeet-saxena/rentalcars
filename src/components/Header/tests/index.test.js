import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from '../index';

afterEach(cleanup);

it('Rendered Snapshot Matches', () => {
    const header = render(<Header />);
    expect(header).toMatchSnapshot();
});
