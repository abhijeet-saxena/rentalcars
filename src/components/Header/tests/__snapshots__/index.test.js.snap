// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered Snapshot Matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM iDAOKJ"
      >
        <header
          class="sc-jSFjdj KhVbt"
        >
          <nav
            class="sc-gKAaRy gXdxYW"
          >
            <a
              href="/rentalcars"
            >
              <img
                alt="Rentalcars.com"
                height="21"
                src="rc-logo.svg"
                width="156"
              />
            </a>
            <div
              class="sc-bdnxRM gjXIaz menu"
              width="auto"
            >
              <span
                aria-label="Prices in Indian Rupees"
                style="color: rgb(255, 255, 255);"
              >
                INR
              </span>
              <img
                alt="Selected Language English"
                src="gb.png"
              />
              <button
                type="button"
              >
                Manage Booking
              </button>
            </div>
            <button
              aria-label="Open Menu"
              class="hamburger"
              type="button"
            >
              <img
                alt=""
                src="hamburger.svg"
              />
            </button>
          </nav>
          <h1>
            Car Hire – Search, Compare & Save
          </h1>
          <div
            class="sc-bdnxRM sc-iCoGMd hoHvTM kyIvOV"
          >
            <h2>
              Let’s find your ideal car
            </h2>
            <div
              class="sc-bdnxRM sc-fujyAs hoHvTM gIcPtD"
            >
              <div
                class="sc-bdnxRM jxqeQX"
                width="auto"
              >
                <input
                  id="diff-location"
                  type="checkbox"
                />
                <label
                  for="diff-location"
                >
                  Drop car off at different location
                </label>
              </div>
              <div
                class="sc-bdnxRM jxqeQX"
                width="auto"
              >
                <input
                  checked=""
                  id="driver-age"
                  type="checkbox"
                />
                <label
                  for="driver-age"
                >
                  Driver aged 30 – 65 ?
                </label>
              </div>
            </div>
            <div
              class="sc-bdnxRM sc-pNWdM hoHvTM fazEXI"
            >
              <div
                class="sc-bdnxRM sc-jrsJWt hoHvTM hLqTlD"
              >
                <input
                  aria-label="Enter Location"
                  placeholder="Pick-Up Location"
                  type="text"
                />
              </div>
              <div
                class="sc-bdnxRM fhotzi"
              >
                <div
                  class="sc-bdnxRM sc-iqAclL hoHvTM eiybFf"
                >
                  <input
                    aria-label="Enter Pickup date"
                    id="pickup-date"
                    name="pickup-date"
                    type="date"
                    value="2021-05-28"
                  />
                  <div
                    class="sc-crzoAE bWDsv"
                  >
                    <select
                      aria-label="Select Pickup Time"
                      name="pickup-time"
                    >
                      <option
                        value="00:00"
                      >
                        00:00
                      </option>
                      <option
                        value="00:30"
                      >
                        00:30
                      </option>
                      <option
                        value="01:00"
                      >
                        01:00
                      </option>
                      <option
                        value="01:30"
                      >
                        01:30
                      </option>
                      <option
                        value="02:00"
                      >
                        02:00
                      </option>
                      <option
                        value="02:30"
                      >
                        02:30
                      </option>
                      <option
                        value="03:00"
                      >
                        03:00
                      </option>
                      <option
                        value="03:30"
                      >
                        03:30
                      </option>
                      <option
                        value="04:00"
                      >
                        04:00
                      </option>
                      <option
                        value="04:30"
                      >
                        04:30
                      </option>
                      <option
                        value="05:00"
                      >
                        05:00
                      </option>
                      <option
                        value="05:30"
                      >
                        05:30
                      </option>
                      <option
                        value="06:00"
                      >
                        06:00
                      </option>
                      <option
                        value="06:30"
                      >
                        06:30
                      </option>
                      <option
                        value="07:00"
                      >
                        07:00
                      </option>
                      <option
                        value="07:30"
                      >
                        07:30
                      </option>
                      <option
                        value="08:00"
                      >
                        08:00
                      </option>
                      <option
                        value="08:30"
                      >
                        08:30
                      </option>
                      <option
                        value="09:00"
                      >
                        09:00
                      </option>
                      <option
                        value="09:30"
                      >
                        09:30
                      </option>
                      <option
                        value="10:00"
                      >
                        10:00
                      </option>
                      <option
                        selected=""
                        value="10:30"
                      >
                        10:30
                      </option>
                      <option
                        value="11:00"
                      >
                        11:00
                      </option>
                      <option
                        value="11:30"
                      >
                        11:30
                      </option>
                      <option
                        value="12:00"
                      >
                        12:00
                      </option>
                      <option
                        value="12:30"
                      >
                        12:30
                      </option>
                      <option
                        value="13:00"
                      >
                        13:00
                      </option>
                      <option
                        value="13:30"
                      >
                        13:30
                      </option>
                      <option
                        value="14:00"
                      >
                        14:00
                      </option>
                      <option
                        value="14:30"
                      >
                        14:30
                      </option>
                      <option
                        value="15:00"
                      >
                        15:00
                      </option>
                      <option
                        value="15:30"
                      >
                        15:30
                      </option>
                      <option
                        value="16:00"
                      >
                        16:00
                      </option>
                      <option
                        value="16:30"
                      >
                        16:30
                      </option>
                      <option
                        value="17:00"
                      >
                        17:00
                      </option>
                      <option
                        value="17:30"
                      >
                        17:30
                      </option>
                      <option
                        value="18:00"
                      >
                        18:00
                      </option>
                      <option
                        value="18:30"
                      >
                        18:30
                      </option>
                      <option
                        value="19:00"
                      >
                        19:00
                      </option>
                      <option
                        value="19:30"
                      >
                        19:30
                      </option>
                      <option
                        value="20:00"
                      >
                        20:00
                      </option>
                      <option
                        value="20:30"
                      >
                        20:30
                      </option>
                      <option
                        value="21:00"
                      >
                        21:00
                      </option>
                      <option
                        value="21:30"
                      >
                        21:30
                      </option>
                      <option
                        value="22:00"
                      >
                        22:00
                      </option>
                      <option
                        value="22:30"
                      >
                        22:30
                      </option>
                      <option
                        value="23:00"
                      >
                        23:00
                      </option>
                      <option
                        value="23:30"
                      >
                        23:30
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  class="sc-bdnxRM sc-iqAclL hoHvTM eiybFf"
                >
                  <input
                    aria-label="Enter Drop date"
                    id="drop-date"
                    name="drop-date"
                    type="date"
                    value="2021-06-01"
                  />
                  <div
                    class="sc-crzoAE bWDsv"
                  >
                    <select
                      aria-label="Select Drop Time"
                      name="drop-time"
                    >
                      <option
                        value="00:00"
                      >
                        00:00
                      </option>
                      <option
                        value="00:30"
                      >
                        00:30
                      </option>
                      <option
                        value="01:00"
                      >
                        01:00
                      </option>
                      <option
                        value="01:30"
                      >
                        01:30
                      </option>
                      <option
                        value="02:00"
                      >
                        02:00
                      </option>
                      <option
                        value="02:30"
                      >
                        02:30
                      </option>
                      <option
                        value="03:00"
                      >
                        03:00
                      </option>
                      <option
                        value="03:30"
                      >
                        03:30
                      </option>
                      <option
                        value="04:00"
                      >
                        04:00
                      </option>
                      <option
                        value="04:30"
                      >
                        04:30
                      </option>
                      <option
                        value="05:00"
                      >
                        05:00
                      </option>
                      <option
                        value="05:30"
                      >
                        05:30
                      </option>
                      <option
                        value="06:00"
                      >
                        06:00
                      </option>
                      <option
                        value="06:30"
                      >
                        06:30
                      </option>
                      <option
                        value="07:00"
                      >
                        07:00
                      </option>
                      <option
                        value="07:30"
                      >
                        07:30
                      </option>
                      <option
                        value="08:00"
                      >
                        08:00
                      </option>
                      <option
                        value="08:30"
                      >
                        08:30
                      </option>
                      <option
                        value="09:00"
                      >
                        09:00
                      </option>
                      <option
                        value="09:30"
                      >
                        09:30
                      </option>
                      <option
                        value="10:00"
                      >
                        10:00
                      </option>
                      <option
                        value="10:30"
                      >
                        10:30
                      </option>
                      <option
                        value="11:00"
                      >
                        11:00
                      </option>
                      <option
                        value="11:30"
                      >
                        11:30
                      </option>
                      <option
                        value="12:00"
                      >
                        12:00
                      </option>
                      <option
                        value="12:30"
                      >
                        12:30
                      </option>
                      <option
                        value="13:00"
                      >
                        13:00
                      </option>
                      <option
                        value="13:30"
                      >
                        13:30
                      </option>
                      <option
                        value="14:00"
                      >
                        14:00
                      </option>
                      <option
                        value="14:30"
                      >
                        14:30
                      </option>
                      <option
                        value="15:00"
                      >
                        15:00
                      </option>
                      <option
                        selected=""
                        value="15:30"
                      >
                        15:30
                      </option>
                      <option
                        value="16:00"
                      >
                        16:00
                      </option>
                      <option
                        value="16:30"
                      >
                        16:30
                      </option>
                      <option
                        value="17:00"
                      >
                        17:00
                      </option>
                      <option
                        value="17:30"
                      >
                        17:30
                      </option>
                      <option
                        value="18:00"
                      >
                        18:00
                      </option>
                      <option
                        value="18:30"
                      >
                        18:30
                      </option>
                      <option
                        value="19:00"
                      >
                        19:00
                      </option>
                      <option
                        value="19:30"
                      >
                        19:30
                      </option>
                      <option
                        value="20:00"
                      >
                        20:00
                      </option>
                      <option
                        value="20:30"
                      >
                        20:30
                      </option>
                      <option
                        value="21:00"
                      >
                        21:00
                      </option>
                      <option
                        value="21:30"
                      >
                        21:30
                      </option>
                      <option
                        value="22:00"
                      >
                        22:00
                      </option>
                      <option
                        value="22:30"
                      >
                        22:30
                      </option>
                      <option
                        value="23:00"
                      >
                        23:00
                      </option>
                      <option
                        value="23:30"
                      >
                        23:30
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <button
                type="button"
              >
                Search
              </button>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM iDAOKJ"
    >
      <header
        class="sc-jSFjdj KhVbt"
      >
        <nav
          class="sc-gKAaRy gXdxYW"
        >
          <a
            href="/rentalcars"
          >
            <img
              alt="Rentalcars.com"
              height="21"
              src="rc-logo.svg"
              width="156"
            />
          </a>
          <div
            class="sc-bdnxRM gjXIaz menu"
            width="auto"
          >
            <span
              aria-label="Prices in Indian Rupees"
              style="color: rgb(255, 255, 255);"
            >
              INR
            </span>
            <img
              alt="Selected Language English"
              src="gb.png"
            />
            <button
              type="button"
            >
              Manage Booking
            </button>
          </div>
          <button
            aria-label="Open Menu"
            class="hamburger"
            type="button"
          >
            <img
              alt=""
              src="hamburger.svg"
            />
          </button>
        </nav>
        <h1>
          Car Hire – Search, Compare & Save
        </h1>
        <div
          class="sc-bdnxRM sc-iCoGMd hoHvTM kyIvOV"
        >
          <h2>
            Let’s find your ideal car
          </h2>
          <div
            class="sc-bdnxRM sc-fujyAs hoHvTM gIcPtD"
          >
            <div
              class="sc-bdnxRM jxqeQX"
              width="auto"
            >
              <input
                id="diff-location"
                type="checkbox"
              />
              <label
                for="diff-location"
              >
                Drop car off at different location
              </label>
            </div>
            <div
              class="sc-bdnxRM jxqeQX"
              width="auto"
            >
              <input
                checked=""
                id="driver-age"
                type="checkbox"
              />
              <label
                for="driver-age"
              >
                Driver aged 30 – 65 ?
              </label>
            </div>
          </div>
          <div
            class="sc-bdnxRM sc-pNWdM hoHvTM fazEXI"
          >
            <div
              class="sc-bdnxRM sc-jrsJWt hoHvTM hLqTlD"
            >
              <input
                aria-label="Enter Location"
                placeholder="Pick-Up Location"
                type="text"
              />
            </div>
            <div
              class="sc-bdnxRM fhotzi"
            >
              <div
                class="sc-bdnxRM sc-iqAclL hoHvTM eiybFf"
              >
                <input
                  aria-label="Enter Pickup date"
                  id="pickup-date"
                  name="pickup-date"
                  type="date"
                  value="2021-05-28"
                />
                <div
                  class="sc-crzoAE bWDsv"
                >
                  <select
                    aria-label="Select Pickup Time"
                    name="pickup-time"
                  >
                    <option
                      value="00:00"
                    >
                      00:00
                    </option>
                    <option
                      value="00:30"
                    >
                      00:30
                    </option>
                    <option
                      value="01:00"
                    >
                      01:00
                    </option>
                    <option
                      value="01:30"
                    >
                      01:30
                    </option>
                    <option
                      value="02:00"
                    >
                      02:00
                    </option>
                    <option
                      value="02:30"
                    >
                      02:30
                    </option>
                    <option
                      value="03:00"
                    >
                      03:00
                    </option>
                    <option
                      value="03:30"
                    >
                      03:30
                    </option>
                    <option
                      value="04:00"
                    >
                      04:00
                    </option>
                    <option
                      value="04:30"
                    >
                      04:30
                    </option>
                    <option
                      value="05:00"
                    >
                      05:00
                    </option>
                    <option
                      value="05:30"
                    >
                      05:30
                    </option>
                    <option
                      value="06:00"
                    >
                      06:00
                    </option>
                    <option
                      value="06:30"
                    >
                      06:30
                    </option>
                    <option
                      value="07:00"
                    >
                      07:00
                    </option>
                    <option
                      value="07:30"
                    >
                      07:30
                    </option>
                    <option
                      value="08:00"
                    >
                      08:00
                    </option>
                    <option
                      value="08:30"
                    >
                      08:30
                    </option>
                    <option
                      value="09:00"
                    >
                      09:00
                    </option>
                    <option
                      value="09:30"
                    >
                      09:30
                    </option>
                    <option
                      value="10:00"
                    >
                      10:00
                    </option>
                    <option
                      selected=""
                      value="10:30"
                    >
                      10:30
                    </option>
                    <option
                      value="11:00"
                    >
                      11:00
                    </option>
                    <option
                      value="11:30"
                    >
                      11:30
                    </option>
                    <option
                      value="12:00"
                    >
                      12:00
                    </option>
                    <option
                      value="12:30"
                    >
                      12:30
                    </option>
                    <option
                      value="13:00"
                    >
                      13:00
                    </option>
                    <option
                      value="13:30"
                    >
                      13:30
                    </option>
                    <option
                      value="14:00"
                    >
                      14:00
                    </option>
                    <option
                      value="14:30"
                    >
                      14:30
                    </option>
                    <option
                      value="15:00"
                    >
                      15:00
                    </option>
                    <option
                      value="15:30"
                    >
                      15:30
                    </option>
                    <option
                      value="16:00"
                    >
                      16:00
                    </option>
                    <option
                      value="16:30"
                    >
                      16:30
                    </option>
                    <option
                      value="17:00"
                    >
                      17:00
                    </option>
                    <option
                      value="17:30"
                    >
                      17:30
                    </option>
                    <option
                      value="18:00"
                    >
                      18:00
                    </option>
                    <option
                      value="18:30"
                    >
                      18:30
                    </option>
                    <option
                      value="19:00"
                    >
                      19:00
                    </option>
                    <option
                      value="19:30"
                    >
                      19:30
                    </option>
                    <option
                      value="20:00"
                    >
                      20:00
                    </option>
                    <option
                      value="20:30"
                    >
                      20:30
                    </option>
                    <option
                      value="21:00"
                    >
                      21:00
                    </option>
                    <option
                      value="21:30"
                    >
                      21:30
                    </option>
                    <option
                      value="22:00"
                    >
                      22:00
                    </option>
                    <option
                      value="22:30"
                    >
                      22:30
                    </option>
                    <option
                      value="23:00"
                    >
                      23:00
                    </option>
                    <option
                      value="23:30"
                    >
                      23:30
                    </option>
                  </select>
                </div>
              </div>
              <div
                class="sc-bdnxRM sc-iqAclL hoHvTM eiybFf"
              >
                <input
                  aria-label="Enter Drop date"
                  id="drop-date"
                  name="drop-date"
                  type="date"
                  value="2021-06-01"
                />
                <div
                  class="sc-crzoAE bWDsv"
                >
                  <select
                    aria-label="Select Drop Time"
                    name="drop-time"
                  >
                    <option
                      value="00:00"
                    >
                      00:00
                    </option>
                    <option
                      value="00:30"
                    >
                      00:30
                    </option>
                    <option
                      value="01:00"
                    >
                      01:00
                    </option>
                    <option
                      value="01:30"
                    >
                      01:30
                    </option>
                    <option
                      value="02:00"
                    >
                      02:00
                    </option>
                    <option
                      value="02:30"
                    >
                      02:30
                    </option>
                    <option
                      value="03:00"
                    >
                      03:00
                    </option>
                    <option
                      value="03:30"
                    >
                      03:30
                    </option>
                    <option
                      value="04:00"
                    >
                      04:00
                    </option>
                    <option
                      value="04:30"
                    >
                      04:30
                    </option>
                    <option
                      value="05:00"
                    >
                      05:00
                    </option>
                    <option
                      value="05:30"
                    >
                      05:30
                    </option>
                    <option
                      value="06:00"
                    >
                      06:00
                    </option>
                    <option
                      value="06:30"
                    >
                      06:30
                    </option>
                    <option
                      value="07:00"
                    >
                      07:00
                    </option>
                    <option
                      value="07:30"
                    >
                      07:30
                    </option>
                    <option
                      value="08:00"
                    >
                      08:00
                    </option>
                    <option
                      value="08:30"
                    >
                      08:30
                    </option>
                    <option
                      value="09:00"
                    >
                      09:00
                    </option>
                    <option
                      value="09:30"
                    >
                      09:30
                    </option>
                    <option
                      value="10:00"
                    >
                      10:00
                    </option>
                    <option
                      value="10:30"
                    >
                      10:30
                    </option>
                    <option
                      value="11:00"
                    >
                      11:00
                    </option>
                    <option
                      value="11:30"
                    >
                      11:30
                    </option>
                    <option
                      value="12:00"
                    >
                      12:00
                    </option>
                    <option
                      value="12:30"
                    >
                      12:30
                    </option>
                    <option
                      value="13:00"
                    >
                      13:00
                    </option>
                    <option
                      value="13:30"
                    >
                      13:30
                    </option>
                    <option
                      value="14:00"
                    >
                      14:00
                    </option>
                    <option
                      value="14:30"
                    >
                      14:30
                    </option>
                    <option
                      value="15:00"
                    >
                      15:00
                    </option>
                    <option
                      selected=""
                      value="15:30"
                    >
                      15:30
                    </option>
                    <option
                      value="16:00"
                    >
                      16:00
                    </option>
                    <option
                      value="16:30"
                    >
                      16:30
                    </option>
                    <option
                      value="17:00"
                    >
                      17:00
                    </option>
                    <option
                      value="17:30"
                    >
                      17:30
                    </option>
                    <option
                      value="18:00"
                    >
                      18:00
                    </option>
                    <option
                      value="18:30"
                    >
                      18:30
                    </option>
                    <option
                      value="19:00"
                    >
                      19:00
                    </option>
                    <option
                      value="19:30"
                    >
                      19:30
                    </option>
                    <option
                      value="20:00"
                    >
                      20:00
                    </option>
                    <option
                      value="20:30"
                    >
                      20:30
                    </option>
                    <option
                      value="21:00"
                    >
                      21:00
                    </option>
                    <option
                      value="21:30"
                    >
                      21:30
                    </option>
                    <option
                      value="22:00"
                    >
                      22:00
                    </option>
                    <option
                      value="22:30"
                    >
                      22:30
                    </option>
                    <option
                      value="23:00"
                    >
                      23:00
                    </option>
                    <option
                      value="23:30"
                    >
                      23:30
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <button
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
